src/main/java/ch/njol/skript/classes/registry/RegistryClassInfo.java

```diff
@@ -16,15 +16,47 @@
  */
 public class RegistryClassInfo<R extends Keyed> extends ClassInfo<R> {
 
+	/**
+	 * @param registryClass The registry class
+	 * @param registry The registry
+	 * @param codeName The name used in patterns
+	 * @param languageNode The language node of the type
+	 */
 	public RegistryClassInfo(Class<R> registryClass, Registry<R> registry, String codeName, String languageNode) {
-		this(registryClass, registry, codeName, languageNode, new EventValueExpression<>(registryClass));
+		this(registryClass, registry, codeName, languageNode, new EventValueExpression<>(registryClass), true);
+	}
+
+	/**
+	 * @param registryClass The registry class
+	 * @param registry The registry
+	 * @param codeName The name used in patterns
+	 * @param languageNode The language node of the type
+	 * @param registerComparator Whether a default comparator should be registered for this registry's classinfo
+	 */
+	public RegistryClassInfo(Class<R> registryClass, Registry<R> registry, String codeName, String languageNode, boolean registerComparator) {
+		this(registryClass, registry, codeName, languageNode, new EventValueExpression<>(registryClass), false);
```

Comment:
```suggestion
		this(registryClass, registry, codeName, languageNode, new EventValueExpression<>(registryClass), registerComparator);
```

Reply:
I have no idea why I put that lol,  whoops